Fix remaining D105 checks: Missing docstring in magic method

### Body

Part of https://github.com/apache/airflow/issues/10742. Recently we enable [D105](https://docs.astral.sh/ruff/rules/undocumented-magic-method/) pydocstyle rule, however we still have some modules which do not follow D105 rule yet. It would be nice if the community help with add missing docstrings to the magic/[special methods](https://docs.python.org/3/reference/datamodel.html#special-method-names)

### Easy way to find all linting problem into the module
1. Remove corresponding line to particular module into the [`pyproject.toml`](https://github.com/apache/airflow/blob/main/pyproject.toml#L1385) in [tool.ruff.lint.per-file-ignores]
2. Run ruff via [pre-commit](https://github.com/apache/airflow/blob/main/contributing-docs/08_static_code_checks.rst#pre-commit-hooks) hook 

```console
❯ pre-commit run ruff --all-files
Run 'ruff' for extremely fast Python linting.............................Failed
- hook id: ruff
- exit code: 1

airflow/models/dag.py:198:9: D105 Missing docstring in magic method
airflow/models/dag.py:765:9: D105 Missing docstring in magic method
airflow/models/dag.py:768:9: D105 Missing docstring in magic method
airflow/models/dag.py:775:9: D105 Missing docstring in magic method
airflow/models/dag.py:778:9: D105 Missing docstring in magic method
airflow/models/dag.py:781:9: D105 Missing docstring in magic method
airflow/models/dag.py:797:9: D105 Missing docstring in magic method
airflow/models/dag.py:801:9: D105 Missing docstring in magic method
airflow/models/dag.py:2424:9: D105 Missing docstring in magic method
airflow/models/dag.py:3489:9: D105 Missing docstring in magic method
airflow/models/dag.py:3510:9: D105 Missing docstring in magic method
airflow/models/dag.py:3630:9: D105 Missing docstring in magic method
airflow/utils/weight_rule.py:43:9: D105 Missing docstring in magic method
Found 13 errors.
```

> [!TIP]
> Feel free to ask a suggestion in slack channels `#contributors` or `#new-contributors`
> if you stack with choose better docstring

---

## Airflow Core
- [x] `airflow/configuration.py`
- [ ] `airflow/exceptions.py`
- [ ] `airflow/plugins_manager.py`
- [ ] `airflow/providers_manager.py`

### [airflow.models](https://github.com/apache/airflow/tree/main/airflow/models)
- [ ] `airflow/models/abstractoperator.py`
- [ ] `airflow/models/baseoperator.py`
- [ ] `airflow/models/connection.py`
- [ ] `airflow/models/dag.py`
- [ ] `airflow/models/dagrun.py`
- [ ] `airflow/models/dagwarning.py`
- [ ] `airflow/models/dataset.py`
- [ ] `airflow/models/expandinput.py`
- [ ] `airflow/models/log.py`
- [ ] `airflow/models/mappedoperator.py`
- [ ] `airflow/models/param.py`
- [ ] `airflow/models/pool.py`
- [ ] `airflow/models/renderedtifields.py`
- [ ] `airflow/models/serialized_dag.py`
- [ ] `airflow/models/slamiss.py`
- [ ] `airflow/models/taskfail.py`
- [ ] `airflow/models/taskinstance.py`
- [ ] `airflow/models/tasklog.py`
- [ ] `airflow/models/taskmixin.py`
- [ ] `airflow/models/variable.py`
- [ ] `airflow/models/xcom.py`
- [ ] `airflow/models/xcom_arg.py`

### [airflow.utils](https://github.com/apache/airflow/tree/main/airflow/utils)
- [ ] `airflow/utils/context.py`
- [ ] `airflow/utils/db.py`
- [ ] `airflow/utils/log/secrets_masker.py`
- [ ] `airflow/utils/operator_resources.py`
- [ ] `airflow/utils/sqlalchemy.py`
- [ ] `airflow/utils/state.py`
- [ ] `airflow/utils/task_group.py`
- [ ] `airflow/utils/timeout.py`
- [ ] `airflow/utils/trigger_rule.py`
- [ ] `airflow/utils/types.py`
- [ ] `airflow/utils/weight_rule.py`

### [airflow.ti_deps](https://github.com/apache/airflow/tree/main/airflow/ti_deps)
- [ ] `airflow/ti_deps/deps/base_ti_dep.py`
- [ ] `airflow/ti_deps/deps/task_not_running_dep.py`
- [ ] `airflow/ti_deps/deps/valid_state_dep.py`

### [airflow.decorators](https://github.com/apache/airflow/tree/main/airflow/decorators)
- [ ] `airflow/decorators/base.py`
- [ ] `airflow/decorators/setup_teardown.py`

### [airflow.io](https://github.com/apache/airflow/tree/main/airflow/io)
- [ ] `airflow/io/path.py`
- [ ] `airflow/io/store/__init__.py`

### [airflow.metrics](https://github.com/apache/airflow/tree/main/airflow/metrics)
- [ ] `airflow/metrics/protocols.py`
- [ ] `airflow/metrics/validators.py`

### [airflow.sensors](https://github.com/apache/airflow/tree/main/airflow/sensors)
- [ ] `airflow/sensors/base.py`
- [ ] `airflow/sensors/external_task.py`

### [airflow.callbacks](https://github.com/apache/airflow/tree/main/airflow/callbacks)
- [ ] `airflow/callbacks/callback_requests.py`

### [airflow.cli](https://github.com/apache/airflow/tree/main/airflow/cli)
- [ ] `airflow/cli/commands/task_command.py`

### [airflow.datasets](https://github.com/apache/airflow/tree/main/airflow/datasets)
- [ ] `airflow/datasets/__init__.py`

### [airflow.executors](https://github.com/apache/airflow/tree/main/airflow/executors)
- [ ] `airflow/executors/base_executor.py`

### [airflow.kubernetes](https://github.com/apache/airflow/tree/main/airflow/kubernetes)
- [ ] `airflow/kubernetes/pre_7_4_0_compatibility/secret.py`

### [airflow.timetables](https://github.com/apache/airflow/tree/main/airflow/timetables)
- [ ] `airflow/timetables/events.py`

### [airflow.triggers](https://github.com/apache/airflow/tree/main/airflow/triggers)
- [ ] `airflow/triggers/base.py`

---

## Provider [amazon](https://github.com/apache/airflow/tree/main/airflow/providers/amazon)
- [ ] `airflow/providers/amazon/aws/exceptions.py`
- [ ] `airflow/providers/amazon/aws/executors/ecs/utils.py`
- [ ] `airflow/providers/amazon/aws/hooks/ecr.py`
- [ ] `airflow/providers/amazon/aws/utils/connection_wrapper.py`

## Provider [databricks](https://github.com/apache/airflow/tree/main/airflow/providers/databricks)
- [ ] `airflow/providers/databricks/hooks/databricks.py`
- [ ] `airflow/providers/databricks/hooks/databricks_base.py`
- [ ] `airflow/providers/databricks/operators/databricks_repos.py`

## Provider [cncf.kubernetes](https://github.com/apache/airflow/tree/main/airflow/providers/cncf/kubernetes)
- [ ] `airflow/providers/cncf/kubernetes/secret.py`
- [ ] `airflow/providers/cncf/kubernetes/utils/delete_from.py`

## Provider [apache](https://github.com/apache/airflow/tree/main/airflow/providers/apache)
- [x] `airflow/providers/apache/hive/hooks/hive.py`

## Provider [elasticsearch](https://github.com/apache/airflow/tree/main/airflow/providers/elasticsearch)
- [ ] `airflow/providers/elasticsearch/log/es_response.py`

## Provider [fab](https://github.com/apache/airflow/tree/main/airflow/providers/fab)
- [ ] `airflow/providers/fab/auth_manager/models/__init__.py`

## Provider [ftp](https://github.com/apache/airflow/tree/main/airflow/providers/ftp)
- [ ] `airflow/providers/ftp/hooks/ftp.py`

## Provider [google](https://github.com/apache/airflow/tree/main/airflow/providers/google)
- [ ] `airflow/providers/google/cloud/links/dataproc.py`

## Provider [imap](https://github.com/apache/airflow/tree/main/airflow/providers/imap)
- [ ] `airflow/providers/imap/hooks/imap.py`

## Provider [microsoft.psrp](https://github.com/apache/airflow/tree/main/airflow/providers/microsoft/psrp)
- [ ] `airflow/providers/microsoft/psrp/hooks/psrp.py`

## Provider [mongo](https://github.com/apache/airflow/tree/main/airflow/providers/mongo)
- [x] `airflow/providers/mongo/hooks/mongo.py`

## Provider [samba](https://github.com/apache/airflow/tree/main/airflow/providers/samba)
- [ ] `airflow/providers/samba/hooks/samba.py`

## Provider [smtp](https://github.com/apache/airflow/tree/main/airflow/providers/smtp)
- [ ] `airflow/providers/smtp/hooks/smtp.py`

## Provider [ssh](https://github.com/apache/airflow/tree/main/airflow/providers/ssh)
- [ ] `airflow/providers/ssh/hooks/ssh.py`

## Provider [tableau](https://github.com/apache/airflow/tree/main/airflow/providers/tableau)
- [ ] `airflow/providers/tableau/hooks/tableau.py`

### Committer

- [X] I acknowledge that I am a maintainer/committer of the Apache Airflow project.

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -273,10 +273,6 @@ extend-select = [
     "RUF100", # Unused noqa (auto-fixable)
     # We ignore more pydocstyle than we enable, so be more selective at what we enable
     "D101",
-    # We add modules that do not follow the rule `Missing docstring in magic method`
-    # into the `tool.ruff.per-file-ignores`, and should remove it from that list as soon as it follows.
-    # See: https://github.com/apache/airflow/issues/10742
-    "D105",
     "D106",
     "D2",
     "D3",
@@ -364,72 +360,6 @@ combine-as-imports = true
 "tests/providers/qdrant/hooks/test_qdrant.py" = ["E402"]
 "tests/providers/qdrant/operators/test_qdrant.py" = ["E402"]
 "tests/providers/snowflake/operators/test_snowflake_sql.py" = ["E402"]
-
-# All the modules which do not follow D105 yet, please remove as soon as it becomes compatible
-"airflow/callbacks/callback_requests.py" = ["D105"]
-"airflow/cli/commands/task_command.py" = ["D105"]
-"airflow/datasets/__init__.py" = ["D105"]
-"airflow/decorators/base.py" = ["D105"]
-"airflow/decorators/setup_teardown.py" = ["D105"]
-"airflow/exceptions.py" = ["D105"]
-"airflow/executors/base_executor.py" = ["D105"]
-"airflow/io/path.py" = ["D105"]
-"airflow/io/store/__init__.py" = ["D105"]
-"airflow/kubernetes/pre_7_4_0_compatibility/secret.py" = ["D105"]
-"airflow/metrics/protocols.py" = ["D105"]
-"airflow/metrics/validators.py" = ["D105"]
-"airflow/models/abstractoperator.py" = ["D105"]
-"airflow/models/baseoperator.py" = ["D105"]
-"airflow/models/connection.py" = ["D105"]
-"airflow/models/dag.py" = ["D105"]
-"airflow/models/dagrun.py" = ["D105"]
-"airflow/models/dagwarning.py" = ["D105"]
-"airflow/models/dataset.py" = ["D105"]
-"airflow/models/expandinput.py" = ["D105"]
-"airflow/models/log.py" = ["D105"]
-"airflow/models/mappedoperator.py" = ["D105"]
-"airflow/models/param.py" = ["D105"]
-"airflow/models/pool.py" = ["D105"]
-"airflow/models/renderedtifields.py" = ["D105"]
-"airflow/models/serialized_dag.py" = ["D105"]
-"airflow/models/slamiss.py" = ["D105"]
-"airflow/models/taskfail.py" = ["D105"]
-"airflow/models/taskinstance.py" = ["D105"]
-"airflow/models/tasklog.py" = ["D105"]
-"airflow/models/taskmixin.py" = ["D105"]
-"airflow/models/variable.py" = ["D105"]
-"airflow/models/xcom.py" = ["D105"]
-"airflow/models/xcom_arg.py" = ["D105"]
-"airflow/plugins_manager.py" = ["D105"]
-"airflow/providers_manager.py" = ["D105"]
-"airflow/sensors/base.py" = ["D105"]
-"airflow/sensors/external_task.py" = ["D105"]
-"airflow/timetables/events.py" = ["D105"]
-"airflow/triggers/base.py" = ["D105"]
-"airflow/utils/context.py" = ["D105"]
-"airflow/utils/db.py" = ["D105"]
-"airflow/utils/log/secrets_masker.py" = ["D105"]
-"airflow/utils/operator_resources.py" = ["D105"]
-"airflow/utils/sqlalchemy.py" = ["D105"]
-"airflow/utils/state.py" = ["D105"]
-"airflow/utils/task_group.py" = ["D105"]
-"airflow/utils/timeout.py" = ["D105"]
-"airflow/utils/trigger_rule.py" = ["D105"]
-"airflow/utils/types.py" = ["D105"]
-"airflow/utils/weight_rule.py" = ["D105"]
-"airflow/providers/cncf/kubernetes/secret.py" = ["D105"]
-"airflow/providers/cncf/kubernetes/utils/delete_from.py" = ["D105"]
-"airflow/providers/databricks/hooks/databricks.py" = ["D105"]
-"airflow/providers/databricks/hooks/databricks_base.py" = ["D105"]
-"airflow/providers/databricks/operators/databricks_repos.py" = ["D105"]
-"airflow/providers/fab/auth_manager/models/__init__.py" = ["D105"]
-"airflow/providers/ftp/hooks/ftp.py" = ["D105"]
-"airflow/providers/google/cloud/links/dataproc.py" = ["D105"]
-"airflow/providers/imap/hooks/imap.py" = ["D105"]
-"airflow/providers/microsoft/psrp/hooks/psrp.py" = ["D105"]
-"airflow/providers/samba/hooks/samba.py" = ["D105"]
-"airflow/providers/smtp/hooks/smtp.py" = ["D105"]
-"airflow/providers/tableau/hooks/tableau.py" = ["D105"]
 
 # All the test modules which do not follow PT012 yet
 "tests/providers/amazon/aws/hooks/test_base_aws.py" = ["PT012"]

